rdoxygen-design.md

```diff
@@ -0,0 +1,126 @@
+# Doxygen Documentation of C++ Libraries in R
+
+## Design Considerations
+
+- *Features.*  
+
+    * **rdoxygen** allows R package developers to add [Doxygen](http://www.stack.nl/~dimitri/doxygen/index.html) documentation to their package for C++ source code (typically found in `src` or `inst/include`).  
+    
+    * Optionally (*should it be optional?*), this documentation can be accessed by users of the package via R [vignettes](http://r-pkgs.had.co.nz/vignettes.html).
```

Comment:
Yes, I believe vignette integration should be optional. When I created this package I only needed the doxygen output to understand my own program, not a vignette. These are separate demands. Also: Doesn't using vignettes also require some extra dependencies like knitr? KISS

Reply:
Agreed: as they cause additional dependencies, Doxygen vignettes should be optional.